docs/language-usage.rst

```diff
@@ -0,0 +1,354 @@
+sentiment about your product on social media or parse intent from
+customer conversations happening in a call center or a messaging
+app. You can analyze text uploaded in your request or integrate
+with your document storage on Google Cloud Storage.
+
+This is a Beta release of Google Cloud Natural Language API. This
+API is not intended for real-time usage in critical applications.
+
+.. _Google Natural Language: https://cloud.google.com/natural-language/docs/getting-started
+
+Client
+------
+
+:class:`~gcloud.language.client.Client` objects provide a
+means to configure your application. Each instance holds
+both a ``project`` and an authenticated connection to the
+Natural Language service.
+
+For an overview of authentication in ``gcloud-python``, see
+:doc:`gcloud-auth`.
+
+Assuming your environment is set up as described in that document,
+create an instance of :class:`~gcloud.language.client.Client`.
+
+  .. code-block:: python
+
+     >>> from gcloud import language
+     >>> client = language.Client()
+
+Methods
+-------
+
+The Google Natural Language API has three supported methods
+
+- `analyzeEntities`_
+- `analyzeSentiment`_
+- `annotateText`_
+
+and each method uses a `Document`_ for representing text. To
+create a :class:`~gcloud.language.document.Document`,
+
+  .. code-block:: python
+
+     >>> text_content = (
+     ...     'Google, headquartered in Mountain View, unveiled the '
+     ...     'new Android phone at the Consumer Electronic Show.  '
+     ...     'Sundar Pichai said in his keynote that users love '
+     ...     'their new Android phones.')
+     >>> document = client.document(text_content)
+
+By default, the document's language and type are English and
+plain text:
+
+  .. code-block:: python
+
+     >>> document.language
+     'en'
+     >>> document.doc_type
+     'PLAIN_TEXT'
+
+These values can also be specified in the constructor
+
+  .. code-block:: python
+
+     >>> html_content = """\
+     ... <html>
+     ...   <head>
+     ...     <title>Story Time</time>
+     ...   </head>
+     ...   <body>
+     ...     <p>La vaca salt&oacute; sobre la luna.</p>
+     ...   </body>
+     ... </html>
+     ... """
+     >>> document = client.document(html_content,
+     ...                            language='es',
+     ...                            doc_type=language.Document.HTML)
+
+The ``language`` arugment can be either ISO-639-1 or BCP-47 language
+codes; at the time, only English, Spanish, and Japanese are supported.
+However, the ``analyzeSentiment`` method only supports English text.
+
+The document type (``doc_type``) argument can be one of
+:attr:`Document.PLAIN_TEXT <gcloud.language.document.Document.PLAIN_TEXT>` or
+:attr:`Document.HTML <gcloud.language.document.Document.HTML>`.
+
+In addition to supplying the text / HTML content, a document can refer
+to content stored in `Google Cloud Storage`_:
+
+  .. code-block:: python
+
+     >>> document = client.document(bucket='my-text-bucket',
```

Comment:
@jerjou What is the format that the back-end expects for the URI?  The [`Document` spec](https://cloud.google.com/natural-language/reference/rest/v1beta1/Document) doesn't say.

-  Bigquery uses `gs://<bucket_name>/<object_name_or_glob>` for its [source URIs](https://cloud.google.com/bigquery/docs/loading-data-cloud-storage).
- Logging uses `storage.googleapis.com/<bucket>` for its `LogSink` [Cloud Storage destination URIs](https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.sinks#LogSink.FIELDS.destination).

It might be more natural to create a `Blob` instance and hand it to the `Document` factory, which would mean that the user didn't need to care about the URI formats.